spring-restdocs-core/src/main/java/org/springframework/restdocs/cli/ConcatenatingCommandFormatter.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2014-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.restdocs.cli;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * {@link CommandFormatter} which concatenates commands with a given {@code separator}.
+ *
+ * @author Tomasz Kopczynski
+ */
+public final class ConcatenatingCommandFormatter implements CommandFormatter {
+
+	private String separator;
+
+	/**
+	 * Creates a new {@code ConcatenatingCommandFormatter} with a newline character as a separator.
+	 * @return {@code ConcatenatingCommandFormatter} with a newline character as a separator.
+	 */
+	public static ConcatenatingCommandFormatter newlineSeparator() {
+		return new ConcatenatingCommandFormatter(" \\%n ");
+	}
+
+	/**
+	 * Creates a new {@code ConcatenatingCommandFormatter} with a space character as a separator.
+	 * @return {@code ConcatenatingCommandFormatter} with a space character as a separator.
+	 */
+	public static ConcatenatingCommandFormatter spaceSeparator() {
```

Comment:
I think this static method would be better on `CliDocumentation`. I'd be tempted to call it `singleLine` or `singleLineFormat`.